aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -115,15 +119,21 @@ public boolean isStarted() {
      * @return A Completable operation to start the sync engine orchestrator
      */
     @NonNull
-    public Completable start() {
-        // Only start if it's stopped.
-        if (!OrchestratorStatus.STOPPED.equals(status.get())) {
-            return initializationCompletable;
+    public synchronized Completable start() {
+        boolean permitAcquired = acquirePermit(OrchestratorAction.START);
+        // Only start if it's stopped AND if we can get a permit.
+        if (!permitAcquired || !status.compareAndSet(OrchestratorStatus.STOPPED, OrchestratorStatus.STARTING)) {
```

Comment:
Augustus De Morgan sez:

```suggestion
        if (!(permitAcquired && status.compareAndSet(OrchestratorStatus.STOPPED, OrchestratorStatus.STARTING)) {
```